rpc-rs/src/provider.rs

```diff
@@ -335,109 +337,124 @@ impl HostBridge {
-                                    &inv.host_id,
-                                    &s
-                                );
-
-                                if let Some(reply_to) = msg.reply {
-                                    // Errors are published from inside the function, safe to ignore Result
-                                    let _ = publish_invocation_response(
-                                        this.rpc_client(),
-                                        reply_to,
-                                        InvocationResponse {
-                                            invocation_id: inv.id,
-                                            error: Some(s.to_string()),
-                                            ..Default::default()
-                                        },
-                                    )
-                                    .await;
                                 }
                             }
                         }
-                    }
-                    Err(e) => {
-                        error!("Invocation deserialization failure: {}", e.to_string());
-                        if let Some(reply_to) = msg.reply {
-                            if let Err(e) = publish_invocation_response(
-                                this.rpc_client(),
-                                reply_to,
-                                InvocationResponse {
-                                    invocation_id: "invalid".to_string(),
-                                    error: Some(format!("Corrupt invocation: {}", e)),
-                                    ..Default::default()
-                                },
-                            )
-                            .await
-                            {
-                                error!("replying to rpc: {}", e);
+                        Err(e) => {
+                            error!(error = %e, "Invocation deserialization failure");
+                            if let Some(reply_to) = msg.reply {
+                                if let Err(e) = publish_invocation_response(
+                                    this.rpc_client(),
+                                    reply_to,
+                                    InvocationResponse {
+                                        invocation_id: "invalid".to_string(),
+                                        error: Some(format!("Corrupt invocation: {}", e)),
+                                        ..Default::default()
+                                    },
+                                )
+                                .await
+                                {
+                                    error!(error = %e, "error when replying to rpc");
+                                }
                             }
                         }
                     }
                 }
             }
-        });
+            .instrument(tracing::trace_span!("subscribe_rpc", %rpc_topic)),
```

Comment:
If this is creating a single span, I don't think this is the right place to put it. Each received rpc should be its own span

Reply:
Yeah, looking at it again, I think it might be better to manually create this span on each loop